src/control/client/types.go

```diff
@@ -469,3 +469,28 @@ func (acl *AccessControlList) Empty() bool {
 	}
 	return false
 }
+
+// PoolDiscovery represents the basic discovery information for a pool.
+type PoolDiscovery struct {
+	UUID        string // Unique identifier
+	SvcReplicas []int  // Ranks of pool service replicas
+}
+
+// poolDiscoveriesFromPB converts the protobuf ListPoolsResp_Pool structures to
+// PoolDiscovery structures.
+func poolDiscoveriesFromPB(pbPools []*mgmtpb.ListPoolsResp_Pool) []*PoolDiscovery {
```

Comment:
@mjmac is now doing this slightly differently from https://github.com/daos-stack/daos/pull/1480
we probably want to be consistent (although could be done as a follow-up as there may be other areas that need migrating)